File: src/main/java/me/miki/shindo/management/settings/impl/BooleanSetting.java
package me.miki.shindo.management.settings.impl;

import me.miki.shindo.management.settings.config.ConfigOwner;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.settings.Setting;

public class BooleanSetting extends Setting {

    private final boolean defaultValue;
    private boolean toggled;

    public BooleanSetting(TranslateText text, ConfigOwner parent, boolean toggled) {
        super(text, parent);
        this.toggled = toggled;
        this.defaultValue = toggled;
    }

    public BooleanSetting(String name, ConfigOwner parent, boolean toggled) {
        super(name, parent);
        this.toggled = toggled;
        this.defaultValue = toggled;
    }

    @Override
    public void reset() {
        this.toggled = defaultValue;
    }

    public boolean isToggled() {
        return toggled;
    }

    public void setToggled(boolean toggle) {
        this.toggled = toggle;
    }

    public boolean isDefaultValue() {
        return defaultValue;
    }
}

File: src/main/kotlin/me/miki/shindo/management/settings/impl/BooleanSetting.kt
package me.miki.shindo.management.settings.impl

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.settings.Setting
import me.miki.shindo.management.settings.config.ConfigOwner

open class BooleanSetting : Setting {

    private val defaultValue: Boolean
    private var toggled: Boolean

    constructor(text: TranslateText, parent: ConfigOwner, toggled: Boolean) : super(text, parent) {
        this.toggled = toggled
        this.defaultValue = toggled
    }

    constructor(name: String, parent: ConfigOwner, toggled: Boolean) : super(name, parent) {
        this.toggled = toggled
        this.defaultValue = toggled
    }

    override fun reset() {
        toggled = defaultValue
    }

    fun isToggled(): Boolean {
        return toggled
    }

    open fun setToggled(toggle: Boolean) {
        toggled = toggle
    }

    fun isDefaultValue(): Boolean {
        return defaultValue
    }
}
